libs/utils/src/json_stream.ts

```diff
@@ -0,0 +1,78 @@
+function isIterable(value: unknown): value is Iterable<unknown> {
+  return (
+    typeof value === 'object' && value !== null && Symbol.iterator in value
+  );
+}
+
+/**
+ * Stream a JSON-serializable value as a series of strings. In addition to the
+ * standard JSON types, this also supports `Iterable` values such as
+ * `Generator`.
+ *
+ * @example
+ *
+ * ```ts
+ * function* generateLargeArray() {
+ *   for (let i = 0; i < 1_000_000_000; i++) {
+ *     yield i;
+ *   }
+ * }
+ *
+ * const stream = jsonStream({ foo: [1, 2, 3], bar: generateLargeArray() });
+ * const writer = createWriteStream('output.json');
+ * for (const chunk of stream) {
+ *  writer.write(chunk);
+ * }
+ * ```
+ */
+export function* jsonStream(input: unknown, indent = 0): Generator<string> {
```

Comment:
What do you think about making a type for json streamable input values? I suggested one here: https://github.com/votingworks/vxsuite/pull/3017#discussion_r1113453460

Reply:
Tried that, didn't seem to work for reasons that weren't totally clear:

<img width="1387" alt="image" src="https://user-images.githubusercontent.com/1938/221717644-dc153b0a-a2cf-4bac-944a-1345823210cc.png">

```ts
type JsonableWithIterable =
  | null
  | boolean
  | string
  | JsonableWithIterable[]
  | Iterable<JsonableWithIterable>
  | Generator<JsonableWithIterable>
  | { [key: string]: JsonableWithIterable };
```

Reply:
Weird

Reply:
I opted to enforce some type safety at the `jsonStream` call site in [this commit](https://github.com/votingworks/vxsuite/pull/3042/commits/6e4a341416ce80b7c8111844dd696cafd81628e6):

```ts
    jsonStream(
      typedAs<WithArraysAsIterables<CVR.CastVoteRecordReport>>({
        ...castVoteRecordReportMetadata,
        CVR: castVoteRecordGenerator,
      })
    )
```

Reply:
And then I changed it to have `jsonStream` take a type parameter:

```ts
    jsonStream<CVR.CastVoteRecordReport>({
      ...castVoteRecordReportMetadata,
      CVR: castVoteRecordGenerator,
    })
```

Reply:
Nice!